src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1791,7 +1792,9 @@ private bool pInvokeMarshalingRequired(CORINFO_METHOD_STRUCT_* handle, CORINFO_S
 #if DEBUG
                 MethodSignature methodSignature = (MethodSignature)HandleToObject((IntPtr)callSiteSig->pSig);
 
-                MethodDesc stub = _compilation.PInvokeILProvider.GetCalliStub(methodSignature);
+                MethodDesc stub = _compilation.PInvokeILProvider.GetCalliStub(
+                    methodSignature,
+                    MarshalHelpers.IsRuntimeMarshallingEnabled(((MetadataType)HandleToObject(callSiteSig->scope).OwningMethod.OwningType).Module));
```

Comment:
Nit: would be nicer to have this pass a `moduleContext` parameter instead of a bool, same as it's done in crossgen2.